### ISSUE TITLE
support \e[2h \e[2l escape codes

### ISSUE THREAD

**Issue by MathGeniusJodie** [tomszilagyi/zutty#14]
\e[2h and \e[2l escape codes are for locking and unlocking keyboard input respectively
`W [vterm.icc:1441] (Unimplemented) set mode 2
W [vterm.icc:1462] (Unimplemented) reset mode 2
`

**Comment by tomszilagyi** [comment]
This is trivial to support (excellent first issue), but I have to ask out of curiosity: What is your use case / what function, in what program does not work due to this not being implemented?

**Comment by MathGeniusJodie** [comment]
if I allow keyboard input while my script runs the characters I type mangle the output of the script

**Comment by tomszilagyi** [comment]
I think that is expected. Don't type characters while your script runs? (:

**Comment by MathGeniusJodie** [comment]
I'm very clumsy

**Comment by koitsu** [comment]
This is not a terminal emulator problem.  Please see `stty` and `termios` man pages, or [this blog post for newbies](https://blog.nelhage.com/2009/12/a-brief-introduction-to-termios-termios3-and-stty/).

Use `stty -echo` in your script to inhibit printing of input characters.  Use `stty echo` to restore printing of input characters.  Other languages (C, Python, etc.) all have these capabilities implemented in some form -- what you're wanting is the equivalent of `tcsetattr()` where you're changing `c_lflag` to have the `ECHO` bit set (print input characters) or clear (inhibit printing).  [Here's an example in C](https://www.gnu.org/software/libc/manual/html_node/Noncanon-Example.html), though you probably do not want to tinker with `ICANON` at all, and [here's an example in Python](https://docs.python.org/3/library/termios.html#example).

Be aware that if you SIGINT/Ctrl-C your script/program while running, depending on UNIX flavour, you may end up with a shell prompt that no longer prints input (but functionally still works).  Your program can trap SIGINT/Ctrl-C (ex. in bash: `trap INT myfunc` + `myfunc() { stty echo; exit 1 }`) to possibly relieve pain.  However, you cannot trap every possible signal, so there can be cases where you can't restore original behaviour (ex. script is `kill -9`d).  This is why it is generally recommended you do not inhibit printing of characters -- it's generally shunned and you can't handle every error condition for proper terminal clean-up.

In short: all of this is normal. You can thank POSIX and UNIX V7 for all of the above nonsense.
